### PR TITLE
WIP: Fix docs links

### PR DESCRIPTION
**Pull request by RadicalZephyr**
#Motivation

Continuing the work started in #1474 to transition all the intra-doc links in the documentation from hard-coded arbitrary urls to using rustdoc item link resolution.

Also related to #1473.

### CODE REVIEW

**Comment by RadicalZephyr** [comment]
I wanted to put this out there just to make it know that I'm working on this, not ready for merging yet.

I could break these fixes up into separate PRs by sub-crate if that seems desirable from a reviewing standpoint.

File: tokio-fs/src/remove_dir.rs [hunk 1]
```diff
@@ -5,9 +5,7 @@ use std::path::Path;
 
 /// Removes an existing, empty directory.
 ///
-/// This is an async version of [`std::fs::remove_dir`][std]
-///
-/// [std]: https://doc.rust-lang.org/std/fs/fn.remove_dir.html
+/// This is an async version of [`std::fs::remove_dir`](std::fs::remove_dir)
```
Comment (ipetkov) [inline tokio-fs/src/remove_dir.rs]: This is fine as is (no need to go back and fix all occurrences of this), but I think if you just use ``[`some::path`]`` it should link things properly given that the path resolves

Comment (RadicalZephyr) [reply tokio-fs/src/remove_dir.rs]: oh cool, I'll keep that in mind!

**Comment by carllerche** [comment]
Woops, sorry for letting this sit :+1:

**Pull request merged**